nixos/modules/installer/tools/tools.nix

```
@@ -224,12 +224,19 @@ in
         # accidentally delete configuration.nix.
         # system.copySystemConfiguration = true;
 
-        # This value determines the NixOS release from which the default
-        # settings for stateful data, like file locations and database versions
-        # on your system were taken. It's perfectly fine and recommended to leave
-        # this value at the release version of the first install of this system.
-        # Before changing this value read the documentation for this option
-        # (e.g. man configuration.nix or on https://nixos.org/nixos/options.html).
+        # This option defines the first version of NixOS you have installed on this particular machine,
+        # and is used to maintain compatibility with data not managed by NixOS configuration (e.g. databases)
+        # that was created by applications running on older NixOS versions.
+        #
+        # Most users should NEVER change this value after the initial install, for any reason.
+        #
+        # This value does NOT affect the Nixpkgs version your packages and OS are pulled from,
+        # so changing it will NOT upgrade your system.
```

```suggestion
        # so changing it will NOT upgrade your system. Conversely, leaving it unchanged it does **not** mean your system is out of date. This value being older than the release of NixOS you are currently using is normal and expected.
```
I feel like this more directly addresses what people often worry about when considering this option. And yes it's redundant, but that's probably a good thing.